source/concepts/applications.rst

```diff
@@ -204,79 +148,133 @@ are defined under the keys ``A``, ``B``, ``C``, and ``D``.
-     \cos{\frac{1}{2} \big(-\theta_i - \frac{2\pi}{N}\big)}\\
+     \cos{\frac{1}{2} \big(-\theta_p - \frac{2\pi}{N}\big)}\\
     \end{bmatrix}
    \end{align}
 
+.. _user-origin-description:
+
+User origin
+~~~~~~~~~~~
+
+A user-origin is a suggestion to a client of how to position their coordinate
+space (and therefore avatar) relative to server space. This is used by the
+:ref:`radial orient <radial-orient>` server feature.
+
+.. note::
+
+   Any client can add user-origin keys. This can be used, for instance, to
+   prototype alternatives to the radial orient feature without modifying the server.
+
+The user origin describes where the server suggests a given user places the center
+of its client space and how to orient it. The origin is described as a protobuf
+Struct under the key ``user-origin.<PLAYER_ID>`` where ``<PLAYER_ID>`` is the ID
+of the user to whom the suggestion is addressed. The Struct has the following keys:
+
+* ``position`` is the suggested location of the center for the user's client
+  space in the server space;
+* ``rotation`` is a quaternion describing the rotation of the user's client
+  space in the server space.
+
+Clients are free to ignore the user-origin suggestion and locate themselves in
+the server space as they choose.
+
+.. warning::
+
+   Any client can add user-origin keys. If used without due care and
+   responsibility a user in VR could get very nauseous.
+
+As a summary, the user origin is specified as follows in the shared state:
+
+.. code::
+
+   user-origin.<PLAYER_ID>: {
+     position,
+     rotation,
+   }
+
+
+.. _multiplayer-update-index:
+
+Update index
+~~~~~~~~~~~~
+
+If the client needs more precise knowledge of which of its updates have already
+been received and rebroadcast to all clients, it can choose to maintain an
+incrementing count of sent updates and store this in the shared state under
+an ``update.index.<USER_ID>`` key. The client can then compare the remotely
+received updates to this value with its internal count.
+
+
 .. _trajectory-application:
 
 The trajectory application
 --------------------------
 
 In the trajectory application, the server broadcasts molecular structures for
 the clients to display. The molecular structures can be static structures or
-snapshots of a trajectory; the protocol refer to these snapshots as frames. The
+snapshots of a trajectory; the protocol refers to these snapshots as frames. The
 application is agnostic about the frames being generated on-the-fly or being
 pre-calculated.
 
-This application defines a set of fields to describes the semantic of molecular
-systems with the ``FrameData``. It also defines a set of optional commands a
+This application defines a set of fields to describe the semantics of molecular
+systems within the ``FrameData``. It also defines a set of optional commands a
 server can implement to give the clients some control over how the frames are
 streamed. Finally, it defines some interactions with the multiplayer
```

Comment:
This sentence needs rewording, but I'm not sure what it means, so I can't make a suggestion